src/ui/components/post/post_listing.rs

```diff
@@ -1,16 +1,306 @@
+    Ok(p) => Ok(p),
+  }
+}
+
+#[server(SavePostFn, "/serverfn")]
+pub async fn save_post_fn(post_id: i32, save: bool) -> Result<PostResponse, ServerFnError> {
+  let form = SavePost {
+    post_id: PostId(post_id),
+    save,
+  };
+
+  let save_result = save_post(&form).await;
+
+  match save_result {
+    Err(e) => Err(ServerFnError::ServerError(e.to_string())),
+    Ok(p) => Ok(p),
+  }
+}
+
+#[server(BlockUserFn, "/serverfn")]
+pub async fn block_user_fn(
+  person_id: i32,
+  block: bool,
+) -> Result<BlockPersonResponse, ServerFnError> {
+  let form = BlockPerson {
+    person_id: PersonId(person_id),
+    block,
+  };
+
+  let save_result = block_user(&form).await;
+
+  match save_result {
+    Err(e) => Err(ServerFnError::ServerError(e.to_string())),
+    Ok(p) => Ok(p),
+  }
+}
+
+#[server(ReportPostFn, "/serverfn")]
+pub async fn report_post_fn(
+  post_id: i32,
+  reason: String,
+) -> Result<PostReportResponse, ServerFnError> {
+  let form = CreatePostReport {
+    post_id: PostId(post_id),
+    reason,
+  };
+
+  let save_result = report_post(&form).await;
+
+  match save_result {
+    Err(e) => Err(ServerFnError::ServerError(e.to_string())),
+    Ok(p) => Ok(p),
+  }
+}
 
 #[component]
-pub fn PostListing(post_view: MaybeSignal<PostView>) -> impl IntoView {
-  let pv = post_view();
-  let link = format!("post/{}", pv.post.id);
+pub fn PostListing(
+  post_view: MaybeSignal<PostView>,
+  error: RwSignal<Option<String>>,
+) -> impl IntoView {
+  let authenticated = expect_context::<RwSignal<bool>>();
+
+  let post_view = create_rw_signal(post_view().clone());
+
+  let vote_action = create_server_action::<VotePostFn>();
+
+  create_effect(move |_| {
+    error.set(None);
+    match vote_action.value().get() {
+      None => {}
+      Some(Ok(o)) => {
+        post_view.set(o.post_view);
+      }
+      Some(Err(e)) => {
+        error.set(Some(e.to_string()));
+      }
+    }
+  });
+
+  let save_post_action = create_server_action::<SavePostFn>();
+
+  create_effect(move |_| {
+    error.set(None);
+    match save_post_action.value().get() {
+      None => {}
+      Some(Ok(o)) => {
+        post_view.set(o.post_view);
+      }
+      Some(Err(e)) => {
+        error.set(Some(e.to_string()));
+      }
+    }
+  });
+
+  let block_user_action = create_server_action::<BlockUserFn>();
+
+  create_effect(move |_| {
+    error.set(None);
+    match block_user_action.value().get() {
+      None => {}
+      Some(Ok(_o)) => {}
+      Some(Err(e)) => {
+        error.set(Some(e.to_string()));
+      }
+    }
+  });
+
+  let report_post_action = create_server_action::<BlockUserFn>();
+
+  create_effect(move |_| {
+    error.set(None);
+    match report_post_action.value().get() {
+      None => {}
+      Some(Ok(_o)) => {}
+      Some(Err(e)) => {
+        error.set(Some(e.to_string()));
+      }
+    }
+  });
+
   view! {
-    <A href=link class="link">
-      {pv.creator.name}
-      -
-      {pv.post.name}
-    </A>
+    <tr>
```

Comment:
I don't think table rows are the way to handle this, and they don't make sense outside of a table table tag and thead/tbody tag.